### ISSUE TITLE
T_format parameter is not in the SourceQuery

### ISSUE THREAD

**Issue by dsavchenko** [oda-hub/dispatcher-app#461]

**Comment by volodymyrss** [comment]
it's not possible to use the unit and format specification of existing parameters? Need to look back on was done few month ago by @burnout87 (with my comments)

**Comment by burnout87** [comment]
BTW [https://github.com/oda-hub/dispatcher-app/pull/254](254) and #336  were addressing the issue

**Comment by dsavchenko** [comment]
It doesn't produce any issue. But currently I'm doing a modification so that unknown (to dispatcher) parameters will be filtered and the notification/warning will be returned. With these changes, it could lead to some problems.

This parameter is anyway passed by frontend and could probably be used silently somewhere (by some instrument backend?). Need to investigate, how does it work now.
At least it should be among the expected parameters in the standard SourceQuery

**Comment by volodymyrss** [comment]
> Does it make sense?

Yes, the list does not even have to be "additional", it's just a full list for request arguments used to construct Parameter.

**Comment by burnout87** [comment]
> 
> Since the format argument in the request to the dispatcher is not found as a Parameter for any query, perhaps a list of "additional accepted arguments" (maybe we can find a better wording for it) including those arguments that are not Parameter, but are used by other Parameters could be an easy extension of the Parameter class itself.

let me rephrase this part better: this list would be used by the Parameter class itself, if present

**Comment by dsavchenko** [comment]
> I am not too sure what you mean since there are several similar terms here:
> 
>     * [Parameter](https://odahub.io/ontology#Parameter) class
> 
>     * dispatcher request parameter/argument (stored in par_dic)
> 
>     * or even input parameters of back-end
> 
> 
> Which one you propose to add?

I proposed just to add an additional Parameter instance to the SourceQuery parameter_list. With this: T_format is in the list of parameters, when one calls get_instrument_description(). Also, in #462 I propose to remove the names, which are not in the Instrument's queries from par_dic, again, T_format should be in the list of parameters somehow. But I agree, adding it as additional parameter is breaking the logic of that it is not separate

**Comment by dsavchenko** [comment]
> I would rather work on the logic related to the generation of those warnings to ensure the format (eg `T_format`) is accepted, and so no warning is issued.

We agreed to move from the warnings in oda_api to generating warnings in the dispatcher.

> This list, associated to the Parameter itself (`Time` Parameter would come with this list including `T_format`), would be checked before issuing the warning.

But it should also be properly treated in whatever method is used to generate the list of instrument's parameter names. I'm not sure it's not too much work

**Comment by volodymyrss** [comment]
> > I would rather work on the logic related to the generation of those warnings to ensure the format (eg `T_format`) is accepted, and so no warning is issued.
> 
> We agreed to move from the warnings in oda_api to generating warnings in the dispatcher.
> 
> > This list, associated to the Parameter itself (`Time` Parameter would come with this list including `T_format`), would be checked before issuing the warning.
> 
> But it should also be properly treated in whatever method is used to generate the list of instrument's parameter names. I'm not sure it's not too much work

*list of arguments

this is a bit complex but inevitable if format is treated in the dispatcher. `T_format` is tightly coupled to times which use.
